internal/broker/services_test.go

```diff
@@ -115,6 +115,166 @@ func TestServices_Services(t *testing.T) {
+		services, err := servicesEndpoint.Services(context.TODO())
+
+		// then
+		require.NoError(t, err)
+		assert.Len(t, services, 1)
+		assert.Len(t, services[0].Plans, 5)
+
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name", "region"})
+		}
+
+	})
+
+	t.Run("should containt the property 'required' with values [name region] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is true", func(t *testing.T) {
+		// given
+		var (
+			name       = "testServiceName"
+			supportURL = "example.com/support"
+		)
+
+		cfg := broker.Config{
+			EnablePlans:                     []string{"gcp", "azure", "openstack", "aws", "free"},
+			IncludeAdditionalParamsInSchema: true,
+			RegionParameterIsRequired:       true,
+			ExposeSchemaWithRegionRequired:  false,
+		}
+		servicesConfig := map[string]broker.Service{
+			broker.KymaServiceName: {
+				Metadata: broker.ServiceMetadata{
+					DisplayName: name,
+					SupportUrl:  supportURL,
+				},
+			},
+		}
+		servicesEndpoint := broker.NewServices(cfg, servicesConfig, logrus.StandardLogger())
+
+		// when
+		services, err := servicesEndpoint.Services(context.TODO())
+
+		// then
+		require.NoError(t, err)
+		assert.Len(t, services, 1)
+		assert.Len(t, services[0].Plans, 5)
+
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name", "region"})
+		}
+
+	})
+
+	t.Run("should containt the property 'required' with values [name] when ExposeSchemaWithRegionRequired is false and RegionParameterIsRequired is false", func(t *testing.T) {
+		// given
+		var (
+			name       = "testServiceName"
+			supportURL = "example.com/support"
+		)
+
+		cfg := broker.Config{
+			EnablePlans:                     []string{"gcp", "azure", "openstack", "aws", "free"},
+			IncludeAdditionalParamsInSchema: true,
+			RegionParameterIsRequired:       false,
+			ExposeSchemaWithRegionRequired:  false,
+		}
+		servicesConfig := map[string]broker.Service{
+			broker.KymaServiceName: {
+				Metadata: broker.ServiceMetadata{
+					DisplayName: name,
+					SupportUrl:  supportURL,
+				},
+			},
+		}
+		servicesEndpoint := broker.NewServices(cfg, servicesConfig, logrus.StandardLogger())
+
+		// when
+		services, err := servicesEndpoint.Services(context.TODO())
+
+		// then
+		require.NoError(t, err)
+		assert.Len(t, services, 1)
+		assert.Len(t, services[0].Plans, 5)
+
+		assert.Equal(t, name, services[0].Metadata.DisplayName)
+		assert.Equal(t, supportURL, services[0].Metadata.SupportUrl)
+
+		for _, plan := range services[0].Plans {
+			assertPlanContainsPropertyValuesInCreateSchema(t, plan, "required", []string{"name"})
```

Comment:
Do you check if there is only `name` or if the list of required contains `name`?

Reply:
I check if there is only `name`

Reply:
For example, if we pass []string{"name","region"}, we will get that the schema doesn't contain the `region`. For example, if we pass an empty slice, we will get that the schema has more values than we passed.